vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1215,13 +1215,13 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
                             }
                         } else {
                             if (wspFilter.getImportMode(path) == ImportMode.REPLACE) {
-                                importInfo.onDeleted(path);
                                 // check if child is not protected
                                 if (child.getDefinition().isProtected()) {
                                     log.debug("Refuse to delete protected child node: {}", path);
-                                } else if (child.getDefinition().isMandatory()) {
+                                } else if (child.getDefinition().isMandatory() && numChildren <= JcrConstants.NUM_MANDATORY_CHILDREN) {
```

Comment:
I don't think that this requires a constant.
```suggestion
                                } else if (child.getDefinition().isMandatory() && numChildren <= 1) {
```

Reply:
however, I think this check is not correct. you need to count the number of child nodes that match the property definition. just having any other child node is not enough.